rclcpp/include/rclcpp/node_interfaces/node_parameters.hpp

```diff
@@ -84,6 +84,16 @@ class NodeParameters : public NodeParametersInterface
 public:
   RCLCPP_SMART_PTR_ALIASES_ONLY(NodeParameters)
 
+  /// Constructor.
+  /**
+   * If using automatically_declare_parameters_from_overrides, overrides of
+   * get_parameter_overrides(), has_parameter(), declare_parameter() will not
+   * be respected.
+   * If this is an issue, pass false for
+   * automatically_declare_parameters_from_overrides and invoke
+   * perform_automatically_declare_parameters_from_overrides() manually after
+   * construction.
+   */
```

Comment:
The new `perform_automatically_declare_parameters_from_overrides()` is protected though, does the suggestion apply only when extending `NodeParameters()`?

Reply:
Yes, only when extending it, which is the case if you were to override one of the aforementioned virtual methods.